netx/httptransport/http3transport.go

```diff
@@ -13,6 +14,21 @@ type HTTP3Dialer interface {
 	Dial(network, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error)
 }
 
+// HTTP3ContextDialer is a dialer for HTTP3 transport using Context.
+type HTTP3ContextDialer interface {
+	DialContext(ctx context.Context, network, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error)
+}
+
+// HTTP3WrapperDialer is a HTTP3Dialer that wraps a ContextDialer
+type HTTP3WrapperDialer struct {
+	Dialer HTTP3ContextDialer
+}
+
+// Dial implements HTTP3Dialer.Dial
+func (d HTTP3WrapperDialer) Dial(network, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+	return d.Dialer.DialContext(context.Background(), network, host, tlsCfg, cfg)
+}
+
```

Comment:
I believe we should take a simpler approach here: 

```Go
// HTTP3Dialer is the definition of dialer for HTTP3 transport assumed by this package.
type HTTP3Dialer interface {
	DialContext(ctx context.Context, network, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error)
}
```

and then modify all the places we were using Dial to call DialContext with the proper context instead.

I believe you introduced the wrapper to preserve backwards compatibility with existing code that was already using `HTTP3Dialer`. Because we don't make API stability guarantees in probe-engine, I would recommend to just refactor existing users of `HTTP3Dialer` so we pass the context as the first argument. This allows us to reduce the overall [optional complexity](http://www.catb.org/~esr/writings/taoup/html/ch13s01.html#id2961759) of the codebase, which is ~always a good idea.

Reply:
There is actually no `DialContext` API for the http3.Roundtripper, you can only assign a custom `Dial` method when creating a new HTTP3 Transport. That's why I built this Wrapper.

Reply:
Ah, that's a bit surprising! I assumed they also had added support in `http3.RoundTripper`. Then, I think it's fine to keep the wrapper around for the time being. (Maybe it makes sense to document why there's the wrapper?)

Reply:
Yes, for sure, I will add an explanation.